src/routes/blog/post/remix-3-whats-changing-and-why-it-matters/+page.markdoc

```diff
@@ -0,0 +1,144 @@
+
+But in May 2025, the team announced Remix 3, an unexpected rebuild from the ground up, meant to rethink how web apps are built. They showed early previews at Remix Jam 2025, and while the final release is planned for 2026, the goals are already clear. Remix 3 breaks away from React entirely and rebuilds around web standards.
+
+It's a big step. The framework now runs directly on the Fetch API, not Node's request/response system. It can run on any JavaScript runtime; Node, Deno, Bun, or Cloudflare Workers, without adapters. It still supports the same development patterns developers liked in Remix 2, but without depending on React's lifecycle or its virtual DOM. The team calls this a move toward being "closer to the web itself."
+
+### Why the Remix team chose to rebuild in v3
+
+When Remix started, React was the natural foundation. Its component model, rendering flow, and community gave Remix an instant head start. But over time, that foundation began to limit what the team could do.
+
+By 2024, the React ecosystem had grown complex. Developers had to manage new patterns like Server Components, `use server` and `use client` boundaries, Suspense, and a flood of new hooks. Next.js, the most visible React framework, kept evolving quickly, but often at the cost of simplicity.
+
+Remix 2, meanwhile, had thinned out. Most of its core logic had moved into React Router, and the extra layer Remix added on top didn't feel worth maintaining. The team decided to stop chasing React's roadmap and instead design something simpler that would work the same way anywhere JavaScript runs.
+
+This rebuild also reflects a return to the web's own strengths. Browsers already know how to handle navigation, forms, and requests. The Fetch API, URL, Request, and Response objects are now available on both the server and client. Remix 3 builds directly on these features instead of wrapping them in extra abstractions. The goal is to make web development less about "framework rules" and more about the platform that's already there.
+
+### The six principles guiding Remix 3
+
+The team described six principles that define Remix 3's direction:
+
+1. Model-first development. Write code that's easy for both people and tools to understand. Each route's intent should be clear about what data it loads, what actions it performs, and how it fits into the overall app. This makes it easier to maintain and to reason about, whether by a developer or an AI tool assisting with code.
+2. Build on web APIs. Use standards like Fetch, Request, Response, and FormData directly. This means a Remix 3 app written for Node can also run on Cloudflare Workers or Deno without code changes.
+3. Religious about runtime. Avoid compile-time "magic." Features should work at runtime, without a custom compiler or code generator. You can still use TypeScript or JSX, but what you write is close to what actually runs.
```

Comment:
We might need rewrite here. First sentence sounds like a weird start.

Also I dont think JS has compilation, more of a build step, bundling..